### PR TITLE
[KeyVault] Fix issue with PATCH method

### PR DESCRIPTION
**Pull request by seankane-msft**
I noticed this issue while fixing #16858 , on PATCH methods, properties would be erased if they are not explicitly set. This fixes that by setting generated values as `nil` if they are not explicitly specified.

### CODE REVIEW

File: sdk/keyvault/azsecrets/CHANGELOG.md [hunk 1]
```diff
@@ -1,12 +1,13 @@
 # Release History
 
-## 0.4.1 (Unreleased)
+## 0.5.0 (Unreleased)
```
Comment (jhendrixMSFT) [inline sdk/keyvault/azsecrets/CHANGELOG.md:3]: For preview, we've been trying to follow the pattern that breaking changes bump the minor version.  This doesn't look to be breaking, at least for compile time.  Is this being considered breaking for runtime?

Comment (seankane-msft) [reply sdk/keyvault/azsecrets/CHANGELOG.md:3]: In runtime I think so, the request body will change with this code change.

Comment (jhendrixMSFT) [reply sdk/keyvault/azsecrets/CHANGELOG.md:3]: I see.  Then should the comment be moved to the `Breaking Changes` category?  It almost makes me think that we could benefit from a `Runtime Changes` category to differentiate between compile-time and runtime breaking changes (maybe we just note this is runtime only in the comment).

Comment (seankane-msft) [reply sdk/keyvault/azsecrets/CHANGELOG.md:3]: Agree that it should be under breaking changes. I don't think we'll have a lot of changes that fall under a "Breaking Change but only at runtime" and for now should just note it in the changelog entry.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**